app/create_file.py

```diff
@@ -1 +1,56 @@
-# write your code here
+from datetime import datetime
+from sys import argv
+from os import mkdir, chdir
+
+filename = ""
+directories = []
+
+if "-f" in argv:
+    index = 0
+
+    for _ in range(len(argv) - 1):
+        index += 1
+        if argv[_: _ + 1] == ["-f"]:
+            break
+
+    filename = argv[index]
+
+    if "-d" in argv:
+        index2 = 0
+
+        for _ in range(len(argv) - 1):
+            index2 += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index2:index - 1]
+else:
+    filename = "file.txt"
+
+    if "-d" in argv:
+        index = 0
+
+        for _ in range(len(argv) - 1):
+            index += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index:len(argv)]
```

Comment:
- According to DRY (Do Not Repeat Yourself) principle we shouldn't repeat same logic in our code. Here we have repeated logic for getting the `-d` argument values. If we have the same logic for something, it's a good practice to make a function for that.

We can check for arguments separately to divide our code.
- Make a if statement for `-f` arg to look for a passed value and set `filename = "file.txt"` if not found.
- Make a if statement for `-d` arg to look for a passed value. In that case remember to stop argv iteration if it reaches value that is other flag (for example `-f`) and not the directory name